tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -294,6 +294,21 @@ impl<F, T> Format<F, T> {
     /// See [`Pretty`].
     ///
     /// Note that this requires the "ansi" feature to be enabled.
+    ///
+    /// # Options
+    ///
+    /// [`Format::with_ansi`] can be used to disable ANSI formatting in event formatting. However,
+    /// a field formatter must be manually provided to avoid ANSI in the formatting of parent
+    /// spans, like so:
+    ///
+    /// ```
+    /// tracing_subscriber::fmt()
+    ///    .pretty()
+    ///    .with_ansi(false)
+    ///    .fmt_fields(format::PrettyFields::new().with_ansi(false))
+    ///    // ... other settings ...
+    ///    .init();
+    /// ```
     #[cfg(feature = "ansi")]
     #[cfg_attr(docsrs, doc(cfg(feature = "ansi")))]
```

Comment:
can we remove the "ansi" feature flag now that this doesn't _depend_ on ANSI formatting? we would have to change it to default to no-ANSI formatting when the `ansi` feature flag is disabled, and update the code so that all the use of `Style`s is feature flagged, like the other formatters.

it would be fine to do this in a follow-up PR after this change lands.

Reply:
I would love to do that! If it's alright, I'd like to do that in a follow-up PR, as I suspect it'll be slightly involved. If it works for you, feel free to create and assign an issue for me :slightly_smiling_face: